ISISNeutronMuon/MDANSE
# Instrument profiles prototype

**Description of work**
The most difficult part of the analysis input at the moment is the q-vector definition. This PR creates a list of parameter profiles, which will initialise resolution and q vectors to the same input every time, which should normally correspond to the parameters of a specific neutron instrument.

Comment: At some point this library https://github.com/oerc0122/ResConvLib will solve the problems with instrument parameters. For now, it will be good for the users to have an option of saving their most commonly used settings.

**Fixes**
1. Added an instrument tab for viewing and editing profiles,
2. Added a settings file with some initial instrument profiles,
3. The job tab has a combo box for instrument selection,
4. The resolution and q vector widgets get their initial values from the instrument profile. These can still be changed manually by the user.
5. Existing definitions can be changed and new ones can be added by the user. These will be stored in a separate TOML file in ~/.mdanse.

**To test**
Load a trajectory and run a Dynamic Incoherent Structure Factor calculation. Change the instrument profile and run the same analysis again. The resolution and q-vector definitions should correspond to those from the instrument profile.
Save a script of the previous analysis. It should contain the correct parameters, without actually referencing the instrument definitions.
Change the parameters of an instrument and save changes. Restart MDANSE_GUI and confirm that the changes have been saved. Delete the ~/.mdanse/InstrumentDefinitions.toml and start MDANSE_GUI again. The original parameters should be available again.
